open-cluster-management/backlog#4679 self mgd hub

cadawson opened the pull request.

cadawson committed:
open-cluster-management/backlog#4679 self mgd hub

swopebe commented on manage_cluster/import.adoc:
```diff
@@ -5,8 +5,7 @@ You can import clusters from different Kubernetes cloud providers.
 
-NOTE: A hub cluster cannot manage _any_ other hub cluster;
-you must import an existing cluster.
+*Note:* A hub cluster cannot manage _any_ other hub cluster. It is configured to automatically import and self-manage itself. You do not need to manually import the hub cluster.
```
@cadawson I need to add to this because of this issue. https://github.com/open-cluster-management/backlog/issues/5693

I don't think this is a note. This is a pretty big conversation and should actually be part of the authoritative documentation... please see if the edits I add work for you and the dev team. Feel free to move around or edit.

swopebe committed:
adding content for self-manage issue

related to  if a hub is removed and reimported, the customer will need to add the local-cluster:true label.  Using the active PR to avoid conflicts and get reviewed simultaneously.

@rokej and @jamestalton does this work?

swopebe committed:
open-cluster-management/backlog#6060 another spot.

@cadawson this was documented twice in the import doc after the move because it came up a lot at IBM and when we got to Red Hat, so I placed it is both spots.

Now that the capability is there, you may decide we only need it in the task. Feel free after the devs approve to remove one of these if you see fit or rearrange it. I was assigned the issue, so I just added to your PR.

swopebe commented on manage_cluster/import_cli.adoc:
```diff
@@ -9,20 +9,22 @@ Follow this procedure to import from the CLI.
+*Important:* A hub cluster cannot manage a different hub cluster. A hub cluster is set up to automatically import and manage itself. You do not have to manually import a hub cluster to manage itself. 
+
+However, if you remove a hub cluster and try to import it again, you need to add the `local-cluster:true` label.
 
```
With line 12-14, can we validate issue: https://github.com/open-cluster-management/backlog/issues/5693

showeimer commented on install/install_connected.adoc:
```diff
@@ -121,7 +121,7 @@ spec:
 +
-*Important:* The `local-cluster` namespace is used for the imported self-managed hub cluster. You must not have a local-cluster namespace on you cluster prior to installing.
+*Important:* The `local-cluster` namespace is used for the imported self-managed hub cluster. You must not have a local-cluster namespace on you cluster prior to installing. After the `local-cluster` namespace is created for the hub cluster, anyone who has access to the `local-cluster` namespace is automatically granted _cluster administrator_ access. Ensure that anyone who is granted access to t ...
 +
```
@jpweiser @ckandag Does this read a little strangely?   While the information is true, I think our recommendation is that **no one** should be granted permissions directly to the `local-cluster` namespace because of the privilege escalation to ClusterAdmin; if that is the desire then just give the user ClusterAdmin directly.  Thoughts?

jpweiser replied:
I agree.

showeimer commented on manage_cluster/import_gui.adoc:
```diff
@@ -160,3 +159,5 @@ Complete the following procedure to remove an imported cluster and the `open-clu
 . From the _Clusters_ page, find your imported cluster in the table.
 . Click *Actions* > *Detach cluster* to remove your cluster from management.
+
+*Note:* The default setting for the hub cluster is to import itself and manage itself. You can change this setting by following the steps to change the `disableHubSelfManagement` value to `true`, as described in link:install_connected.adoc#installing-while-connected-online[Installing while connected online]. If the value is left at its default value of `false`, the hub cluster reimports itself af ...
```
Seems good; the re-import may not be immediate; because it will happen at the next reconcile of the MultiClusterHub controller

jpweiser reviewed (changes requested):
See https://github.com/open-cluster-management/rhacm-docs/pull/1424/files#r514325196

cadawson committed:
open-cluster-management/backlog#4679 rev cmts

cadawson committed:
open-cluster-management/backlog#4679 rev cmts

cadawson commented:
Updated the wording around this @jpweiser and @showeimer . Let me know how this works. Thanks!

cadawson committed:
open-cluster-management/backlog#6474 Add trigger

cadawson committed:
open-cluster-management/backlog#6474 Add trigger

cadawson committed:
open-cluster-management/backlog#4679 Add trigger

cadawson committed:
open-cluster-management/backlog#4679 Add trigger

cadawson committed:
open-cluster-management/backlog#4679 Add trigger

cadawson committed:
open-cluster-management/backlog#4679 Add trigger

cadawson commented:
Merging to add the updates.
